glide-core/src/client/value_conversion.rs

```diff
@@ -379,6 +380,68 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        ExpectedReturnType::GeoSearchReturnType(dist, hash, coord) => match value {
+            Value::Array(array) => {
+                if !dist && !hash && !coord {
+                    return Ok(Value::Array(array));
+                }
+
+                let mut converted_array = Vec::with_capacity(array.len());
+                let num_true = dist as usize + hash as usize + coord as usize;
+                for item in array.iter() {
+                    if let Value::Array(inner_array) = item {
+                        if let Some((name, rest)) = inner_array.split_first() {
+                            if rest.len() != num_true {
+                                return Err((
+                                    ErrorKind::TypeError,
+                                    "Response couldn't be converted to GeoSearchReturn type",
+                                    format!("(response was {:?})", array),
+                                )
+                                    .into());
+                            }
+                            let rest = rest
+                                .iter()
+                                .map(|v| match v {
+                                    Value::Array(coord) => {
+                                        if coord.len() != 2 {
+                                            return Err((
+                                                ErrorKind::TypeError,
+                                                "Inner Array must contain exactly two elements, longitude and latitude",
+                                            )
+                                                .into());
+                                        }
+                                        let longitude = convert_to_expected_type(
+                                            coord[0].clone(),
+                                            Some(ExpectedReturnType::Double),
+                                        )?;
+                                        let latitude = convert_to_expected_type(
+                                            coord[1].clone(),
+                                            Some(ExpectedReturnType::Double),
+                                        )?;
+                                        Ok(Value::Array(vec![longitude, latitude]))
+                                    }
+                                    Value::BulkString(dist) => convert_to_expected_type(
```

Comment:
`dist` is the input argument.  Consider a different variable name. 
```suggestion
                                    Value::BulkString(dist_or_hash) => convert_to_expected_type(
```

Reply:
does this work for both dist (double) and hash (integer)

Reply:
this is just for sidt, hash is an integer for both resp2/3